src/vm/object.cpp

```diff
@@ -221,6 +221,11 @@ TypeHandle Object::GetGCSafeTypeHandleIfPossible() const
         pMTToCheck = thElem.GetMethodTable();
     }
 
+    if (pMTToCheck == g_pFreeObjectMethodTable)
```

Comment:
This should be done as the first thing at top right after `GetGCSafeMethodTable`